### DISCUSSION TITLE
Sunton 2.8" CYD Display (ESP32-2432S028R) with uTouch since Tasmota 14.0

### DISCUSSION THREAD

**Discussion by patpommi** [arendst/Tasmota#21808] [Q&A]
Hello!
I have the CYD.
Display and touch works fine at Vers. 13.4

But after flashing with Vers. 14.0 or 14.1 only the display works.
While displaycalibrate the display reacts on touches.
But after touching the last cross, the calibration stops at this screen without the cross.
![grafik](https://github.com/user-attachments/assets/7de14365-d67f-44e0-8c94-47c79adf059a)

 
Neither the succesful nor failed screen is shown.
Console after calibration:
15:35:57.194 MQT: stat/tasmota_31D608/RESULT = {"DisplayRotate":0}
15:36:05.656 TS : Calibration (-42,-20) - (-42,-20) - (-42,-20) - (-42,-20)
15:36:05.659 BRY: Exception> 'divzero_error' - division by zero
15:36:05.660 stack traceback:
15:36:05.661    <unknown source>: in function `calc_geometry`
15:36:05.661    <unknown source>: in function `finish`
15:36:05.662    <unknown source>: in function `do_next_corner`
15:36:05.663    <unknown source>: in function `<lambda>`
15:36:05.664    <unknown source>: in function `run_deferred`
15:36:05.664    <unknown source>: in function `event`

Console after startup:
00:00:00.080 BRY: Berry initialized, RAM used 5341 bytes
00:00:00.498 UT : XPT2046
00:00:00.498 DSP: Configured display 'ILI9341'
00:00:00.507 Project tasmota - 14.1 Version 14.1.0(release-lvgl-haspmota)-3_0_0(2024-06-03T11:56:02)
00:00:00.514 LVG: LVGL initialized
00:00:00.600 TAP: Loaded Tasmota App 'DisplayCalibrate.tapp'
00:00:01.242 BRY: Successfully loaded 'autoexec.be'
00:00:02.098 WIF: Connecting to AP1 HaB Channel 1 BSSId 7C:FF:4D:9C:6A:4A in mode HT40 as tasmota-31D608-5640...
00:00:04.498 WIF: Connected
00:00:04.758 HTP: Web server active on tasmota-31D608-5640 with IP address 192.168.
…

Configuration:
 
![grafik](https://github.com/user-attachments/assets/d5d1ee1b-29d2-4c10-b31f-e78a702bfb16)

Display.ini is from here without changes:
https://github.com/arendst/Tasmota/blob/development/tasmota/displaydesc/ILI9341_xpt_display.ini

Or with changing this line:
:UTI,XPT2046,S1,*,-1,*    #In my display.ini there are two stars (*). They are not shown here.
Or like this:
:UTI,XPT2046,S1,33,-1,36
All three variations don’t work and effects the same as described above.

We have tried much variations. Only the display works.
Can someone help with the right settings?
Thanks, Ralf

**Comment by sfromis** [comment]
What you post does not suggest that you completed the calibration. On the contrary, an error message like `BRY: Exception> 'divzero_error' - division by zero` looks like it failed, hence no chance of touch working.

**Reply by patpommi** [reply]
Yes, the calibration stops at the point described.
My question now is, what did I do wrong at the configuration? Or which settings do I have to change and how?

**Reply by s-hadinger** [reply]
Your logs show that the touchscreen does not work since it provided the same coordinates for all 4 points

**Reply by patpommi** [reply]
Thanks for the quik answer.
The touchscreen works.
When i flash Vers. 13.4 again, dispaly an touchscreen works fine.

**Reply by Jumagiera** [reply]
i have the same issue  with 14.1. touch always return 0,0.
I’m using exactly the same board und configuration

how c an i get a working 13.4 with hasp?

**Reply by Jumagiera** [reply]
found 13.4. I was confuse because i’m working with a esp32-s3 in parallel.
im 13.4 touch works fine. so it seams that the xpt2046 support in the universal driver is broken

**Reply by patpommi** [reply]
I fear the same about the uTouch-driver for XPT2046.

The CYD works well with 13.4, but some of the LVGL objects do not yet work in 13.4.
So it only really makes sense from 14.0 onwards.
And what is possible in terms of displays on 14.0/14.1 is really great!

**Comment by patpommi** [comment]
Fixed!
Look here:
https://github.com/arendst/Tasmota/issues/21810

**Reply by snelweg** [reply]
still not working here.

**Reply by patpommi** [reply]
Have you read this?
https://github.com/arendst/Tasmota/issues/21810
